compose/ui/ui-text/src/desktopTest/kotlin/androidx/compose/ui/text/DesktopParagraphIntegrationTest.kt

```diff
@@ -0,0 +1,4226 @@
+
+        assertThat(paragraph.isLineEllipsized(0)).isTrue()
+        assertThat(paragraph.getLineStart(0)).isEqualTo(0)
+        assertThat(paragraph.getLineEnd(0)).isEqualTo(text.length)
+    }
+
+    @Test
+    fun lineHeight_inSp() {
+        val text = "abcdefgh"
+        val fontSize = 20f
+        // Make the layout 4 lines
+        val layoutWidth = text.length * fontSize / 4
+        val lineHeight = 30f
+
+        val paragraph =
+            simpleParagraph(
+                text = text,
+                style = TextStyle(fontSize = fontSize.sp, lineHeight = lineHeight.sp),
+                width = layoutWidth,
+            )
+
+        assertThat(paragraph.lineCount).isEqualTo(4)
+        // First/last line is influenced by top/bottom padding
+        for (i in 1 until paragraph.lineCount - 1) {
+            val actualHeight = paragraph.getLineHeight(i)
+            // In the sample_font.ttf, the height of the line should be
+            // fontSize + 0.2f * fontSize(line gap)
+            assertWithMessage("line number $i").that(actualHeight).isEqualTo(lineHeight)
+        }
+    }
+
+    @Test
+    fun lineHeight_InEm() {
+        val text = "abcdefgh"
+        val fontSize = 20f
+        // Make the layout 4 lines
+        val layoutWidth = text.length * fontSize / 4
+        val lineHeight = 1.5f
+
+        val paragraph =
+            simpleParagraph(
+                text = text,
+                style = TextStyle(fontSize = fontSize.sp, lineHeight = lineHeight.em),
+                width = layoutWidth,
+            )
+
+        assertThat(paragraph.lineCount).isEqualTo(4)
+        // First/last line is influenced by top/bottom padding
+        for (i in 1 until paragraph.lineCount - 1) {
+            val actualHeight = paragraph.getLineHeight(i)
+            // In the sample_font.ttf, the height of the line should be
+            // fontSize + 0.2f * fontSize(line gap)
+            assertWithMessage("line number $i").that(actualHeight).isEqualTo(lineHeight * fontSize)
+        }
+    }
+
+    @Test
+    fun testAnnotatedString_setFontSizeOnWholeText() {
+        with(defaultDensity) {
+            val text = "abcde"
+            val fontSize = 20.sp
+            val fontSizeInPx = fontSize.toPx()
+            val spanStyle = SpanStyle(fontSize = fontSize)
+            val paragraphWidth = fontSizeInPx * text.length
+
+            val paragraph =
+                simpleParagraph(
+                    text = text,
+                    spanStyles = listOf(AnnotatedString.Range(spanStyle, 0, text.length)),
+                    width = paragraphWidth,
+                )
+
+            // Make sure there is only one line, so that we can use getLineRight to test fontSize.
+            assertThat(paragraph.lineCount).isEqualTo(1)
+            // Notice that in this test font, the width of character equals to fontSize.
+            assertThat(paragraph.getLineWidth(0)).isEqualTo(fontSizeInPx * text.length)
+        }
+    }
+
```

Comment:
When you fix failures on CI/Linux, I will run tests on Windows, to check if everything is okay

Reply:
Checked on Windows